### DISCUSSION TITLE
Error Margin Insufficient

### DISCUSSION THREAD

**Discussion by hargums** [conor19w/Binance-Futures-Trading-Bot#30] [General]
Please reply sir, i had best strategy also,  i struggle to run Live_Bot.py, but running from app.py is working fine,  it said Error Margin Insufficient, example :
i add order size in app.py 2.5 it working fine running smooth, but if i running from Live_Bot.py it become error : Error Margin Insufficient

this is my config file, im using tester binance account that had $15 only for testing purpose, i had best strategi that maybe suitable for everyone, the strategy is in the last post

##fill in your API keys here to be accessed by other scripts
API_KEY = 'my api'
API_SECRET = 'my api'

################## settings, these are very strategy dependant ensure you have enough data for your chosen strategy ##################################
order_Size = 11
leverage = 20
buffer = '2 days ago'  ## Buffer of candle sticks be careful if you don't provide enough the bot will throw an error
Interval = '5m'  ##candle sticks you want to trade
Max_Number_Of_Trades = 3 ## How many positions we can have open at once
use_trailing_stop = 0 ##If on we will use our TP value as the Activation price for a trailing stop loss
trailing_stop_callback = 0.8  ##trailing stop percent, this is .1% range is [.1% - 5%] .ie [0.1 - 5]

## New vars needed for the gui, running script from terminal will also need these now
strategy = 'tripleEMAStochasticRSIATR'
TP_choice = 'x (ATR)'
SL_choice = 'x (ATR)'
SL_mult = 1.169
TP_mult = 2.3

##Trade All Coins if True, can also specify a list of coins to trade instead. Example: symbol = ['ETHUSDT','BTCUSDT'] & set Trade_All_Coins = False
Trade_All_Coins = False
symbol = ['C98USDT', 'ADAUSDT', '1000XECUSDT', 'EOSUSDT', 'CHZUSDT', 'MATICUSDT', 'CTKUSDT', 'FILUSDT', 'AXSUSDT', 'ANTUSDT', 'ZECUSDT', 'GTCUSDT', 'COMPUSDT', 'KSMUSDT', 'COTIUSDT', 'FLMUSDT', 'AUDIOUSDT', 'CTKUSDT', 'PEOPLEUSDT', 'ARPAUSDT', 'DOGEUSDT', '1000SHIBUSDT', 'MANAUSDT', 'CVCUSDT', 'TRXUSDT', 'OGNUSDT', 'RSRUSDT', 'WOOUSDT', 'THETAUSDT', 'RENUSDT', 'OGNUSDT', 'GALAUSDT']  ## If Trade_All_Coins is False then we list the coins we want to trade here, otherwise the bot will automatically get all coins and trade them

my strategy just for bullish market

def tripleEMAStochasticRSIATR(Close, High, Low, Trade_Direction, SL, TP, TP_choice, SL_choice, EMA50, EMA14, EMA8, fastd, fastk, current_index):
    ##buy signal ini sudah digantikan trade aslinya itu 1 dan 0 lagi fastk nya, atur SL 3 atk, tp 1.3 atk
    if   (Close[current_index] > EMA8[current_index] < EMA14[current_index] < EMA50[current_index]) and (fastk[current_index]<18) and (fastd[current_index]<18):
        Trade_Direction = 1

    elif (Close[current_index] < EMA8[current_index] > EMA14[current_index] > EMA50[current_index])  and (fastk[current_index]>81) and (fastd[current_index]>81):
        Trade_Direction = 0

    stop_loss_val, take_profit_val = SetSLTP(-99, -99, Close, High, Low, Trade_Direction, SL, TP, TP_choice, SL_choice, current_index)
    return Trade_Direction, stop_loss_val, take_profit_val

**Comment by conor19w** [comment]
I Changed the Bot now so it will read your inputs properly, Previously the order size needed to be in decimal that's why you were getting an insufficient margin error. (11 was the equivalent of 1100% of your account and then leverage brought that up to 20x1100% so thats why you were getting the error.)
